### ISSUE TITLE
Appender - support all appends

### ISSUE THREAD

**Issue by ryanhamilton** [duckdb/duckdb-java#110]
I'm currently using batch insert statements to insert data to duckdb from java. As specified in the docs:

> The preferred method for bulk inserts is to use the [Appender](https://duckdb.org/docs/api/java#appender) due to its higher performance. However, when using the Appender is not possbile, the batch writer is available as alternative.

Unfortunately some of my data are dates / times and nested arrays.
Could duckdb_append_date/duckdb_append_time be exposed in java?
and a method of adding arrays?
It would be nice to be able to say ``appender.append(new int[]{3,2,1});`` or similar for each data type.

### DuckDB C API

```
duckdb_state [duckdb_append_date](https://duckdb.org/docs/api/c/appender#duckdb_append_date)(duckdb_appender appender, duckdb_date value);
duckdb_state [duckdb_append_time](https://duckdb.org/docs/api/c/appender#duckdb_append_time)(duckdb_appender appender, duckdb_time value);
duckdb_state [duckdb_append_interval](https://duckdb.org/docs/api/c/appender#duckdb_append_interval)(duckdb_appender appender, duckdb_interval value);
duckdb_state [duckdb_append_blob](https://duckdb.org/docs/api/c/appender#duckdb_append_blob)(duckdb_appender appender, const void *data, idx_t length);
duckdb_state [duckdb_append_data_chunk](https://duckdb.org/docs/api/c/appender#duckdb_append_data_chunk)(duckdb_appender appender, duckdb_data_chunk chunk);
```

**Comment by Colory** [comment]
append_default is also needed.

**Comment by staticlibs** [comment]
Yes, `appendNull`, `appendDefault` and `appendArray` are planned to be added soon, will look also into `appendDate`.

**Comment by wernerdaehn** [comment]
All datatypes please, also including Map, Struct, Union, BLOB!

Kind of annoying if you are supposed to use the Appender and it supports only half the data types, isn't it?

**Comment by staticlibs** [comment]
Almost everything from this issue (except `MAP` and `UNION`) is implemented now in the reworked `Appender` in `1.3.2.0`. So I am closing this issue, please reopen if needed.

**Comment by rayokota** [comment]
Thanks @staticlibs , created https://github.com/duckdb/duckdb-java/issues/307 to track MAP and UNION support.
